docs/proposals/disk-affinity.md

```diff
@@ -0,0 +1,161 @@
+# Proposal: Supporting disk affinity/anti-affinity feature 
+
+Last updated: 18.01.2022
+
+## Abstract
+
+This proposal contains approach for supporting disk affinity/anti-affinity feature.
+
+## Background
+
+Current lvg storageclass cannot mount physical disks to PVs with the awareness of the requesting pods. So, it's possible
+for different applications to share the same disk.
+CSI should provide a way to support:
+* mount PVs for the same pod on the same disk (disk affinity);
+* do not mount PVs on the disks that already have some PVs from other specific pods (disk antiaffinity).
+
+## Proposal
+
+Usually for affinity/antiaffinity topology feature is used. But the minimal domain, on which this feature operates, is nodes, not disks.
+So we should use a different approach for this one. Another approach, which is proposed, is to use annotations for pointing out how to use disk affinity/antiaffinity:
+affinity.drives.csi-baremetal.dell.com/type: <pod-bound> - mount PVs for the same pod on the same disk;
+antiaffinity.drives.csi-baremetal.dell.com/type: <pod-label> - specify disk antiaffinity type: currently there is only one option - by pod labels;  
```

Comment:
Ho will this work with other applications which don't have integration with proposed APIs?

Reply:
Can you please give some examples of this kind of applications, which cannot support this API?

Reply:
Some applications might be running by customer and we don't have control over it. The main concern here how will we guarantee that disk will in exclusive usage by application with proposed APIs.

Reply:
If customer/we don't know which applications to select by the labels - we can make the following:
Mark drives/ac as dedicated, e.g. by adding the special label:
affinity.csi-baremetal.dell.com/dedicated: nginx
and then just use annotations at pods:
affinity.drives.csi-baremetal.dell.com/type: drive-label
affinity.drives.csi-baremetal.dell.com/drives: nginx

Reply:
Added new type and options to approach